tests/test_CharArrayNullTerm.cpp

```diff
@@ -0,0 +1,78 @@
+#define BOOST_TEST_MODULE Aggregate_Well_Data
+
+#include <boost/test/unit_test.hpp>
+
+#include <opm/output/eclipse/CharArrayNullTerm.hpp>
+
+// =====================================================================
+
+BOOST_AUTO_TEST_SUITE(AChar8)
+
+BOOST_AUTO_TEST_CASE (Basic_Operations)
+{
+    // Default Constructor
+    {
+        const auto s = Opm::RestartIO::Helpers::CharArrayNullTerm<8>{};
```

Comment:
If the string is too long?

Reply:
> If the string is too long?

Then you get [truncation](https://github.com/bska/opm-common/blob/19c0033122008ba5b6032fd528472735daba6a1a/opm/output/eclipse/CharArrayNullTerm.hpp#L87-L99).

I'll add a test case for it.